### PR TITLE
Add admission webhook validation for Path based on path type

### PR DESCRIPTION
**Pull request by ccfishk**
**What type of PR is this?**

/kind feature

**Does this PR introduce a user-facing change?**:
```release-note
 * HTTP Path match values are validated for PathMatchExact, PathMatchPrefix type
```

**What this PR does / why we need it**:
Webhook validatation of HTTPPathMatch.Value for Exact and Prefix Type

**Which issue(s) this PR fixes**:

Fixes #
Issue https://github.com/kubernetes-sigs/gateway-api/issues/873

### CODE REVIEW

File: apis/v1alpha2/validation/httproute.go [hunk 1]
```diff
@@ -28,6 +31,9 @@ var (
 	repeatableHTTPRouteFilters = []gatewayv1a2.HTTPRouteFilterType{
 		gatewayv1a2.HTTPRouteFilterExtensionRef,
 	}
+
+	invalidPathSequences = []string{"//", "/./", "/../", "%2f", "%2F"}
```
Comment (howardjohn) [inline apis/v1alpha2/validation/httproute.go:35]: Could add `#` as well https://github.com/envoyproxy/envoy/security/advisories/GHSA-r222-74fw-jqr9

Comment (howardjohn) [reply apis/v1alpha2/validation/httproute.go:35]: @yangminzhu may have some additional ones

File: apis/v1alpha2/validation/httproute_test.go [hunk 2]
```diff
@@ -389,3 +389,44 @@ func TestValidateHTTPBackendUniqueFilters(t *testing.T) {
...
+		{
+			name: "invalid httpRoute prefix",
+			path: &gatewayv1a2.HTTPPathMatch{
+				Type:  pkgutils.PathMatchTypePtr("Prefix"),
+				Value: utilpointer.String("/."),
+			},
+			errCount: 1,
+		},
+		{
+			name: "valid httpRoute Exact",
```
Comment (robscott) [inline apis/v1alpha2/validation/httproute_test.go:408]:
```suggestion
			name: "invalid httpRoute Exact",
```

File: apis/v1alpha2/validation/httproute.go [hunk 3]
```diff
@@ -93,3 +99,36 @@ func validateHTTPBackendUniqueFilters(ref []gatewayv1a2.HTTPBackendRef, path *fi
...
+func validateHTTPPathMatch(path *gatewayv1a2.HTTPPathMatch, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if path.Type == nil {
+		return append(allErrs, field.Required(fldPath.Child("pathType"), "pathType must be specified"))
+	}
+
+	switch *path.Type {
+	case gatewayv1a2.PathMatchExact, gatewayv1a2.PathMatchPrefix:
+		if !strings.HasPrefix(*path.Value, "/") {
```
Comment (robscott) [inline apis/v1alpha2/validation/httproute.go:113]: Although defaulting should generally remove the possibility of a nil value here, I think it's still best to add a nil check before trusting there is a value.

File: apis/v1alpha2/validation/httproute.go (hunk 1 above)
Comment (ccfishk) [reply apis/v1alpha2/validation/httproute.go]: https://github.com/kubernetes-sigs/gateway-api/pull/894/commits/a87064c9ac5a3d05ce51af84ae7083ccbf5ed3da

File: apis/v1alpha2/validation/httproute.go (hunk 3 above)
Comment (ccfishk) [reply apis/v1alpha2/validation/httproute.go:117]: https://github.com/kubernetes-sigs/gateway-api/pull/894/commits/a87064c9ac5a3d05ce51af84ae7083ccbf5ed3da

**Comment by robscott** [comment]
Thanks!

/lgtm
/approve

**Pull request merged**